crmsh/utils.py

```diff
@@ -185,7 +185,8 @@ def _get_user_of_host_from_config(host):
     def _guess_user_for_ssh(host: str) -> typing.Tuple[str, str]:
         args = ['ssh']
         args.extend(constants.SSH_OPTION_ARGS)
-        args.extend(['-o', 'BatchMode=yes', host, 'sudo', 'true'])
+        sudo_str = 'sudo ' if userdir.get_sudoer() else ''
+        args.extend(['-o', 'BatchMode=yes', host, sudo_str, 'true'])
```

Comment:
1. When passing arguments as array of strings, space is not needed. Space is the separator of arguments in shell. See `init main(int argc, char **argv)` and `execve(1)`.
2. An empty argument is not the same as non-existing argument. `argv = {"true"}` and  `argv = {"", "true"}` is not the same.

```suggestion
        if userdir.get_sudoer():
            args.extend(['-o', 'BatchMode=yes', host, 'sudo', 'true'])
        else:
            args.extend(['-o', 'BatchMode=yes', host, 'true'])
```